<launch>
  <arg name="map_path"            default=""/>
  <arg name="trajectory_path"     default=""/>
  <arg name="odom_topic"    default="/odom"/>
  <arg name="n_particles"         default="10"/>
  <arg name="model_sensitivity"         default="0.8"/>
  <arg name="model_false_positive_rate" default="0.2"/>

  <!-- Defaults, change only if find it necessary -->
  <arg name="reference_frame_id"  default="base_link"/>
  <arg name="publish_map_to_frame_tf"          default="false"/>
  <arg name="publish_utm_to_map_tf"          default="true"/>

  <node name="relative_offset_node" type="relative_offset_transform" pkg="mapless_mcl_ros" output="screen">
    <remap from="/relative/pose" to="/odometry/offset"/>
    <param name="odom_topic" value="$(arg odom_topic)"/>
  </node>

  <node name="drmcl_node" type="run.py" pkg="mapless_mcl_ros" output="screen">
    <param name="particles"         value="$(arg n_particles)"/>
    <param name="publish_map_to_frame_tf"     value="$(arg publish_map_to_frame_tf)"/>
    <param name="publish_utm_to_map_tf"       value="$(arg publish_utm_to_map_tf)"/>
    <param name="frame"             value="$(arg reference_frame_id)"/>
    <param name="map_path"          value="$(arg map_path)"/>
    <param name="trajectory_path"   value="$(arg trajectory_path)"/>
    <param name="model_sensitivity" value="$(arg model_sensitivity)"/>
    <param name="model_false_positive_rate" value="$(arg model_false_positive_rate)"/>
    <param name="trajectory_path"   value="$(arg trajectory_path)"/>
  </node>

</launch>